.github/workflows/main.yml

```diff
@@ -18,7 +18,7 @@ jobs:
     steps:
 
     - name: Checkout repository
-      uses: actions/checkout@v4
+      uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.1
```

Comment:
These were originally with just a tag. We trust GitHub, so no need to use the hash. We also use tag for Microsoft azure actions, since Microsoft owns GitHub.